projects/plugins/boost/app/modules/cache/page_cache.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Automattic\Jetpack_Boost\Modules\Page_Cache;
+
+use Automattic\Jetpack_Boost\Contracts\Is_Always_On;
+use Automattic\Jetpack_Boost\Contracts\Pluggable;
+
+class Page_Cache implements Pluggable, Is_Always_On {
+	/*
+	 * Sets up the advanced-cache.php file and if that works, adds the WP_CACHE
+	 * define to wp-config.php
+	 * These are used by WordPress to load the caching system before most of
+	 * WordPress is loaded.
+	 */
+	public function setup() {
+		if ( is_admin() ) {
+			require_once ABSPATH . '/wp-admin/includes/file.php';
+			WP_Filesystem();
+			$error = $this->create_advanced_cache();
+			if ( $error === true ) {
+				$this->add_wp_cache_define();
+			}
+			// do something with $error
```

Comment:
Should this run every time an admin page loads?

Reply:
No, it should really only run on the Jetpack Boost page. I'll add a page check.